xcm/xcm-executor/src/lib.rs

```diff
@@ -275,6 +275,10 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				let assets = holding.saturating_take(assets);
 				for asset in assets.assets_iter() {
 					Config::AssetTransactor::check_out(&origin, &asset);
+					ensure!(
+						Config::IsTeleporter::filter_asset_location(&asset, &dest),
+						XcmError::UntrustedTeleportLocation
+					);
```

Comment:
Check should come first before any sort of state mutation happens:

```suggestion
					ensure!(
						Config::IsTeleporter::filter_asset_location(&asset, &dest),
						XcmError::UntrustedTeleportLocation
					);
					Config::AssetTransactor::check_out(&origin, &asset);
```